src/tools/rbd/action/Nbd.cc

```diff
@@ -173,9 +181,12 @@ int execute_detach(const po::variables_map &vm,
     return -EINVAL;
   }
 
-  std::vector<std::string> args;
+  if (device_name.empty() && vm.count(at::SNAPSHOT_ID)) {
+    uint64_t snapid = vm[at::SNAPSHOT_ID].as<uint64_t>();
+    args.push_back("--snap-id");
+    args.push_back(boost::lexical_cast<std::string>(snapid));
```

Comment:
It looks like this converts a string into an integer only to convert it back into a string again?

Reply:
I see.  This can still be simplified though:
```
args.push_back(std::to_string(vm[at::SNAPSHOT_ID].as<uint64_t>()));
```